src/main/java/seedu/address/logic/commands/EditCommand.java

```diff
@@ -92,16 +102,20 @@ public CommandResult execute(Model model) throws CommandException {
      * Creates and returns a {@code Person} with the details of {@code personToEdit}
      * edited with {@code editPersonDescriptor}.
      */
-    private static Person createEditedPerson(Person personToEdit, EditPersonDescriptor editPersonDescriptor) {
+    private static Person createEditedPerson(Person personToEdit, EditPersonDescriptor editPersonDescriptor)
+            throws CommandException {
         assert personToEdit != null;
 
+        Role updatedRole = editPersonDescriptor.getRole().orElse(personToEdit.getRole());
         Name updatedName = editPersonDescriptor.getName().orElse(personToEdit.getName());
         Phone updatedPhone = editPersonDescriptor.getPhone().orElse(personToEdit.getPhone());
         Email updatedEmail = editPersonDescriptor.getEmail().orElse(personToEdit.getEmail());
         Address updatedAddress = editPersonDescriptor.getAddress().orElse(personToEdit.getAddress());
         Set<Tag> updatedTags = editPersonDescriptor.getTags().orElse(personToEdit.getTags());
 
-        return new Person(updatedName, updatedPhone, updatedEmail, updatedAddress, updatedTags);
+        Person person = PersonFactory.createPerson(updatedRole, updatedName, updatedPhone, updatedEmail, updatedAddress,
+                updatedTags, editPersonDescriptor, personToEdit);
```

Comment:
Is this a formatting issue? Should `updatedTags` and so forth be next to the previous line instead?

Reply:
It should. It'll be exceeding 120 line limit if put in single line